Source: Written by Calvin Walton <calvin.walton@kepstin.ca>
Upstream: TODO
Reason: setup.py fails to read some files when system locale is not UTF-8

diff -u -r sure-1.2.7.orig/setup.py sure-1.2.7/setup.py
--- sure-1.2.7.orig/setup.py	2014-06-30 18:00:50.172098300 -0400
+++ sure-1.2.7/setup.py	2014-06-30 18:14:05.367297500 -0400
@@ -17,6 +17,7 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import ast
+import io
 import os
 from setuptools import setup, find_packages
 
@@ -37,12 +38,15 @@
 def read_version():
     """Read version from sure/version.py without loading any files"""
     finder = VersionFinder()
-    finder.visit(ast.parse(local_file('sure', '__init__.py')))
+    finder.visit(ast.parse(local_bytes_file('sure', '__init__.py')))
     return finder.version
 
 
-local_file = lambda *f: \
-    open(os.path.join(os.path.dirname(__file__), *f)).read()
+local_bytes_file = lambda *f: \
+    open(os.path.join(os.path.dirname(__file__), *f), 'rb').read()
+
+local_text_file = lambda *f: \
+    io.open(os.path.join(os.path.dirname(__file__), *f), encoding='utf-8').read()
 
 
 tests_require = ['nose', 'mock']
@@ -53,7 +57,7 @@
           version=read_version(),
           description='utility belt for automated testing in python for python',
           author='Gabriel Falcao',
-          long_description=local_file('README.rst'),
+          long_description=local_text_file('README.rst'),
           author_email='gabriel@nacaolivre.org',
           include_package_data=True,
           url='http://github.com/gabrielfalcao/sure',
